apidef/oas/authentication.go

```diff
@@ -167,6 +167,35 @@ func (ss SecuritySchemes) Import(name string, nativeSS *openapi3.SecurityScheme,
 	return nil
 }
 
+func (ss SecuritySchemes) GetBaseIdentityProvider() (res apidef.AuthTypeEnum) {
+	if len(ss) < 2 {
+		return
+	}
+
+	precedence := map[apidef.AuthTypeEnum]int{
```

Comment:
Consider moving to a global var to avoid allocation per function call. An alternative would be something like `func (apidef.AuthTypeEnum) int` with a switch/return block which can be inlined without allocations

Reply:
👍🏻